aptos-move/aptos-e2e-comparison-testing/src/execution.rs

```diff
@@ -202,152 +233,180 @@ impl Execution {
                 compiled_package_cache_v2,
+                None,
             );
         }
         Ok(())
     }
 
-    fn execute_and_compare(
+    pub(crate) fn execute_and_compare(
         &self,
         cur_version: Version,
-        state: &FakeDataStore,
-        features: &Features,
+        state: FakeDataStore,
         txn_idx: &TxnIndex,
         compiled_package_cache: &HashMap<PackageInfo, HashMap<ModuleId, Vec<u8>>>,
         compiled_package_cache_v2: &HashMap<PackageInfo, HashMap<ModuleId, Vec<u8>>>,
+        debugger: Option<Arc<dyn AptosValidatorInterface + Send>>,
     ) {
         let mut package_cache_main = compiled_package_cache;
         let package_cache_other = compiled_package_cache_v2;
         if self.execution_mode.is_v2() {
             package_cache_main = compiled_package_cache_v2;
         }
         let res_main_opt = self.execute_code(
-            state,
-            features,
+            cur_version,
+            state.clone(),
             &txn_idx.package_info,
             &txn_idx.txn,
             package_cache_main,
+            debugger.clone(),
         );
         if self.execution_mode.is_compare() {
             let res_other_opt = self.execute_code(
+                cur_version,
                 state,
-                features,
                 &txn_idx.package_info,
                 &txn_idx.txn,
                 package_cache_other,
+                debugger.clone(),
             );
-            Self::print_mismatches(cur_version, &res_main_opt.unwrap(), &res_other_opt.unwrap());
+            self.print_mismatches(cur_version, &res_main_opt.unwrap(), &res_other_opt.unwrap());
         } else {
             let res = res_main_opt.unwrap();
             if let Ok(res_ok) = res {
-                println!(
+                self.output_result_str(format!(
                     "version:{}\nwrite set:{:?}\n events:{:?}\n",
                     cur_version, res_ok.0, res_ok.1
-                );
+                ));
             } else {
-                println!(
+                self.output_result_str(format!(
                     "execution error {} at version: {}, error",
                     res.unwrap_err(),
                     cur_version
-                );
+                ));
             }
         }
     }
 
     fn execute_code(
         &self,
-        state: &FakeDataStore,
-        features: &Features,
+        version: Version,
+        state: FakeDataStore,
         package_info: &PackageInfo,
         txn: &Transaction,
         compiled_package_cache: &HashMap<PackageInfo, HashMap<ModuleId, Vec<u8>>>,
+        debugger_opt: Option<Arc<dyn AptosValidatorInterface + Send>>,
     ) -> Option<Result<(WriteSet, Vec<ContractEvent>), VMStatus>> {
         let executor = FakeExecutor::no_genesis();
         let mut executor = executor.set_not_parallel();
-        *executor.data_store_mut() = state.clone();
+        *executor.data_store_mut() = state;
```

Comment:
So we overwrite default data created by `FakeExecutor` with empty state?

Reply:
yes, it happens in the online execution mode